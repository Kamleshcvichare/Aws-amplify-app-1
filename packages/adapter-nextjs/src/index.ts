// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

export { runWithAmplifyServerContext } from './runWithAmplifyServerContext';
export { NextServer } from './types';
