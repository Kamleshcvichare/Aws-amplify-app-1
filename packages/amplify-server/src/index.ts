export { AmplifyServerWorkload } from './amplifyServerPayload';
export { NextAuth, NuxtAuth } from './frameworks';
