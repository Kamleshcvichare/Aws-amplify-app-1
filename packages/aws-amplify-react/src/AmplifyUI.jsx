/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';

import { JS } from 'aws-amplify';

import AmplifyTheme from './AmplifyTheme';

export const Container = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.container);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-container", theme.container)} style={style}>
            {props.children}
        </div>
    )
}

export const FormContainer = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.formContainer);
    return beforeAfter(
        <div className={themedClassName("amplify-form-container", theme.formContainer)} style={style}>
            {props.children}
        </div>
    )
}

export const FormSection = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.formSection);
    return (
        <FormContainer theme={theme}>
            {beforeAfter(
                <div className={themedClassName("amplify-form-section", theme.formSection)} style={style}>
                    {props.children}
                </div>
            )}
        </FormContainer>
    )
}

export const ErrorSection = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.errorSection);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-error-section", theme.errorSection)} style={style}>
            <ErrorSectionContent>
                {props.children}
            </ErrorSectionContent>
        </div>
    )
}

export const ErrorSectionContent = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.errorSectionContent);
    return beforeAfter(
        <span className={themedClassName("amplify-error-section-content", theme.errorSectionContent)} style={style}>
            {props.children}
        </span>
    )
}

export const SectionHeader = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.sectionHeader);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-section-header", theme.sectionHeader)} style={style}>
            <SectionHeaderContent theme={theme}>
                {props.children}
            </SectionHeaderContent>
        </div>
    )
}

export const SectionHeaderContent = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.sectionHeaderContent);
    return beforeAfter(
        <span className={themedClassName("amplify-section-header-content", theme.sectionHeaderContent)} style={style}>
            {props.children}
        </span>
    )
}

export const SectionFooter = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.sectionFooter);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-section-footer", theme.sectionFooter)} style={style}>
            <SectionFooterContent>
                {props.children}
            </SectionFooterContent>
        </div>
    )
}

export const SectionFooterContent = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.sectionFooterContent);
    return beforeAfter(
        <span className={themedClassName("amplify-section-footer-content", theme.sectionFooterContent)} style={style}>
            {props.children}
        </span>
    )
}

export const SectionBody = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.sectionBody);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-section-body", theme.sectionBody)} style={style}>
            {props.children}
        </div>
    )
}

export const ActionRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.actionRow);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-action-row", theme.actionRow)} style={style}>
            {props.children}
        </div>
    )
}

export const FormRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.formRow);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-form-row", theme.formRow)} style={style}>
            {props.children}
        </div>
    )
}

export const InputRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.input);
    const p = JS.objectLessAttributes(props, 'theme');
    return (
        <FormRow theme={theme}>
            {beforeAfter(
                <input
                    {...p}
                    className={themedClassName("amplify-input", theme.input)}
                    style={style}
                />
            )}
        </FormRow>
    )
}

export const RadioRow = (props) => {
    const id = props.id || '_' + props.value;
    const theme = props.theme || AmplifyTheme;
    return (
        <FormRow theme={theme}>
            <Radio
                {...props}
                id={id}
            />
            <Label
                htmlFor={id}
                theme={theme}
            >{props.placeholder}</Label>
        </FormRow>
    )
}

export const Radio = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.radio);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
            <input
                {...p}
                type="radio"
                className={themedClassName("amplify-radio", theme.radio)}
                style={style}
            />
    )
}

export const CheckboxRow = (props) => {
    const id = props.id || '_' + props.name;
    const theme = props.theme || AmplifyTheme;
    return (
        <FormRow theme={theme}>
            <Checkbox
                {...props}
                id={id}
            />
            <Label
                htmlFor={id}
                theme={theme}
            >{props.placeholder}</Label>
        </FormRow>
    )
}

export const Checkbox = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.checkbox);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <input
            {...p}
            type="checkbox"
            className={themedClassName("amplify-checkbox", theme.checkbox)}
            style={style}
        />
    )
}

export const MessageRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <FormRow theme={theme}>
            <MessageContent theme={theme}>{props.children}</MessageContent>
        </FormRow>
    )
}

export const MessageContent = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <span className={themedClassName("amplify-message-content", theme.messageContent)} style={theme.messageContent}>
            {props.children}
        </span>
    )
}

export const ButtonRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    const { className, ...style } = theme.actionRow || {}
    return beforeAfter(
        <div className={themedClassName("amplify-action-row", { className })} style={style}>
            <Button {...props} />
        </div>
    )
}

export const Button = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.button);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <button {...p} className={themedClassName("amplify-button", theme.button)} style={style}>
            <ButtonContent theme={theme}>{props.children}</ButtonContent>
        </button>
    )
}

export const ButtonContent = (props) => {
    const theme = props.theme || AmplifyTheme;
    const { className, ...style } = theme.buttonContent || {}
    return beforeAfter(
        <span className={themedClassName("amplify-button-content", { className })} style={style}>
            {props.children}
        </span>
    )
}

export const SignInButton = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.signInButton);
    const p = JS.objectLessAttributes(props, 'theme');

    return beforeAfter(
        <button {...p} className={themedClassName("amplify-signin-button", theme.signInButton)} style={style}>
            {props.children}
        </button>
    )
}

export const Link = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.a);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <a {...p} className={themedClassName("amplify-a", theme.a)} style={style}>{props.children}</a>
    )
}

export const Label = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.label);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <label {...p} className={themedClassName("amplify-label", theme.label)} style={style}>{props.children}</label>
    )
}

export const Space = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.space);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <span {...p} className={themedClassName("amplify-space", theme.space)} style={style}>{props.children}</span>
    )
}

export const NavBar = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.navBar);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-nav-bar", theme.navBar)} style={style}>
            {props.children}
        </div>
    )
}

export const Nav = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.nav);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-nav", theme.nav)} style={style}>
            {props.children}
        </div>
    )
}

export const NavRight = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.navRight);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-nav-right", theme.navRight)} style={style}>
            {props.children}
        </div>
    )
}

export const NavItem = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.navItem);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <div {...p} className={themedClassName("amplify-nav-item", theme.navItem)} style={style}>
            {props.children}
        </div>
    )
}

export const NavButton = (props) => {
    const theme = props.theme || AmplifyTheme;
    const style = propStyle(props, theme.navButton);
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <button {...p} className={themedClassName("amplify-nav-button", theme.navButton)} style={style}>
            {beforeAfter(
                <span style={theme.navButtonContent}>{props.children}</span>
            )}
        </button>
    )
}

export const beforeAfter = (el) => {
    const style = el.props.style || {};
    const { before, after } = style;
    if (!before && !after) { return el; }

    return (
        <span style={{position: 'relative'}}>
            {before? <span style={before}>{before.content}</span> : null}
            {el}
            {after? <span style={after}>{after.content}</span> : null}
        </span>
    )
}

export const propStyle = (props, themeStyle) => {
    const { id, style } = props;
    const styl = Object.assign({}, style, themeStyle);
    delete styl.className
    if (!id) { return styl; }

    const selector = '#' + id;
    Object.assign(styl, styl[selector]);
    return styl;
}

export const themedClassName = (defaultClassName, theme) => {
    if (!theme || !theme.className) {
        return defaultClassName;
    }
    return `${defaultClassName} ${theme.className || ''}`
}

export const transparent1X1 = 'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==';

export const white1X1 = 'data:image/gif;base64,R0lGODlhAQABAIAAAP7//wAAACH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==';
