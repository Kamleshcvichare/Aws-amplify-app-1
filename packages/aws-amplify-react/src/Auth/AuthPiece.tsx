/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { Component } from 'react';
import { ConsoleLogger as Logger } from '@aws-amplify/core';

import AmplifyTheme from '../Amplify-UI/Amplify-UI-Theme';

export interface IAuthPieceProps {
    authData?: any;
    authState?: string;
    hide?: any;
    onAuthEvent?: any;
    onStateChange?: (state: string, data?) => void;
    track?: () => void;
    theme?: any;
}

export interface IAuthPieceState {
}

export default class AuthPiece<Props extends IAuthPieceProps, State extends IAuthPieceState> extends Component<Props, State> {
    public _validAuthStates: string[];
    public _isHidden: boolean;
    public inputs: any; 

    constructor(props: Props) {
        super(props);

        this.inputs = {};

        this._isHidden = true;
        this._validAuthStates = [];
        this.changeState = this.changeState.bind(this);
        this.error = this.error.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
    }

    // extract username from authData
    usernameFromAuthData() {
        const { authData } = this.props;
        if (!authData) { return ''; }

        let username = '';
        if (typeof authData === 'object') { // user object
            username = authData.user? authData.user.username : authData.username;
        } else {
            username = authData; // username string
        }

        return username;
    }

    errorMessage(err: any) {
        if (typeof err === 'string') { return err; }
        return err.message? err.message : JSON.stringify(err);
    }

    triggerAuthEvent(event: any) {
        const state = this.props.authState;
        if (this.props.onAuthEvent) { this.props.onAuthEvent(state, event); }
    }

    changeState(state: any, data?: any) {
        if (this.props.onStateChange) { this.props.onStateChange(state, data); }

        this.triggerAuthEvent({
            type: 'stateChange',
            data: state
        });
    }

    error(err: any) {
        this.triggerAuthEvent({
            type: 'error',
            data: this.errorMessage(err)
        });
    }

    handleInputChange(evt: any) {
        this.inputs = this.inputs || {};
        const { name, value, type, checked } = evt.target;
        const check_type = ['radio', 'checkbox'].includes(type);
        this.inputs[name] = check_type? checked : value;
        this.inputs['checkedValue'] = check_type? value: null;
    }

    render() {
        if (!this._validAuthStates.includes(this.props.authState)) {
            this._isHidden = true;
            this.inputs = {};
            return null;
        }

        if (this._isHidden) {
            this.inputs = {};
            const { track } = this.props;
            if (track) track();
        }
        this._isHidden = false;

        return this.showComponent(this.props.theme || AmplifyTheme);
    }

    showComponent(theme: any) {
        console.error('You must implement showComponent(theme) and don\'t forget to set this._validAuthStates.');
        return null;
    }
}
