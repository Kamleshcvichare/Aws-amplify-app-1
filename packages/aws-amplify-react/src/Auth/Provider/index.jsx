import React, { Component } from 'react';

import withGoogle from './withGoogle';
import withFacebook from './withFacebook';
import withAmazon from './withAmazon';

export { default as withGoogle, GoogleButton } from './withGoogle';
export { default as withFacebook, FacebookButton } from './withFacebook';
export { default as withAmazon, AmazonButton } from './withAmazon';

export function withFederated(
    Comp,
    { amazon, facebook, google } = {
        amazon: true,
        facebook: true,
        google: true
    }
) {
    let Federated = Comp;
    if (facebook) Federated = withFacebook(Federated);
    if (google) Federated = withGoogle(Federated);
    if (amazon) Federated = withAmazon(Federated);

    return class extends Component {
        render() {
            const federated = this.props.federated || {};
            return <Federated {...this.props} {...federated} />;
        }
    };
}
