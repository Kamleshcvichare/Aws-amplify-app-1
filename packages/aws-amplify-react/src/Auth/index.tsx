/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import * as React from 'react';
import { Component } from 'react';
import { CognitoUser } from 'amazon-cognito-identity-js';

import { UsernameAttributes, FederatedConfig, AuthState } from './common/types';
import Authenticator from './Authenticator';

export { default as Authenticator } from './Authenticator';
export { default as AuthPiece } from './AuthPiece';
export { default as SignIn } from './SignIn';
export { default as ConfirmSignIn } from './ConfirmSignIn';
export { default as SignOut } from './SignOut';
export { default as RequireNewPassword } from './RequireNewPassword';
export { default as SignUp } from './SignUp';
export { default as ConfirmSignUp } from './ConfirmSignUp';
export { default as VerifyContact } from './VerifyContact';
export { default as ForgotPassword } from './ForgotPassword';
export { default as Greetings } from './Greetings';
export { default as FederatedSignIn, FederatedButtons } from './FederatedSignIn';
export { default as TOTPSetup } from './TOTPSetup';
export { default as Loading } from './Loading';

export * from './Provider';

interface AuthenticatorPropsBase {
    federated?: FederatedConfig
    onStateChange: (state: AuthState, data: {}) => void
}

interface AuthenticatorPropsSignedIn extends AuthenticatorPropsBase {
    authState: 'signedIn'
    authData: CognitoUser
}

interface AuthenticatorPropsNotSignedIn extends AuthenticatorPropsBase {
    authState:
        | 'signIn'
        | 'signUp'
        | 'confirmSignIn'
        | 'confirmSignUp'
        | 'forgotPassword'
        | 'requireNewPassword'
        | 'verifyContact'
    authData: null
}

export type AuthenticatorProps =
    | AuthenticatorPropsSignedIn
    | AuthenticatorPropsNotSignedIn

interface State {
    authState?: AuthenticatorProps['authState']
    authData?: AuthenticatorProps['authData']
}

export function withAuthenticator<P extends AuthenticatorProps>(
    Comp: React.ComponentType<P>,
    includeGreetings: boolean = false,
    authenticatorComponents: React.ComponentType[] = [],
    federated: FederatedConfig = null,
    theme: {} = null,
    signUpConfig: {} = {}
): React.ComponentClass<Omit<P, keyof AuthenticatorProps>> {
    return class extends Component<Omit<P, keyof AuthenticatorProps>, State> {
        private authConfig: {
            includeGreetings?: boolean
            authenticatorComponents?: React.ComponentType[],
            federated?: FederatedConfig,
            usernameAttributes?: UsernameAttributes
            theme?: {},
            signUpConfig?: {}
        }

        constructor(props) {
            super(props);

            this.handleAuthStateChange = this.handleAuthStateChange.bind(this);

            this.state = {
                authState: props.authState || null,
                authData: props.authData || null
            };

            this.authConfig = {};

            if (typeof includeGreetings === 'object' && includeGreetings !== null){
                this.authConfig = Object.assign(this.authConfig, includeGreetings)
            } else {
                this.authConfig = {
                    includeGreetings,
                    authenticatorComponents,
                    federated,
                    theme,
                    signUpConfig
                }
            }
        }

        handleAuthStateChange(state, data) {
            this.setState({ authState: state, authData: data });
        }

        render() {
            const { authState, authData } = this.state;
            if (authState === 'signedIn') {
                return (
                    <React.Fragment>
                        { this.authConfig.includeGreetings ? 
                            <Authenticator
                                {...this.props}
                                theme={this.authConfig.theme}
                                federated={this.authConfig.federated || this.props.federated}
                                hideDefault={this.authConfig.authenticatorComponents && this.authConfig.authenticatorComponents.length > 0}
                                signUpConfig={this.authConfig.signUpConfig}
                                usernameAttributes={this.authConfig.usernameAttributes}
                                onStateChange={this.handleAuthStateChange}
                                children={this.authConfig.authenticatorComponents || []}
                            /> : null
                        }
                        <Comp
                            {...this.props}
                            authState={authState}
                            authData={authData}
                            onStateChange={this.handleAuthStateChange}
                        />
                    </React.Fragment>
                );
            }

            return <Authenticator
                {...this.props}
                theme={this.authConfig.theme}
                federated={this.authConfig.federated || this.props.federated}
                hideDefault={this.authConfig.authenticatorComponents && this.authConfig.authenticatorComponents.length > 0}
                signUpConfig={this.authConfig.signUpConfig}
                usernameAttributes={this.authConfig.usernameAttributes}
                onStateChange={this.handleAuthStateChange}
                children={this.authConfig.authenticatorComponents || []}
            />;
        }
    };
}

interface AuthenticatorWrapperProps {
    children: (auth: AuthenticatorWrapperState['auth']) => React.ReactNode
}
interface AuthenticatorWrapperState {
    auth: string
    authData?: CognitoUser
}

export class AuthenticatorWrapper extends Component<AuthenticatorWrapperProps, AuthenticatorWrapperState> {
    constructor(props) {
        super(props);

        this.state = { auth: 'init' };

        this.handleAuthState = this.handleAuthState.bind(this);
    }

    handleAuthState(state, data) {
        this.setState({ auth: state, authData: data });
    }

    render() {
        return (
            <div>
                <Authenticator {...this.props} onStateChange={this.handleAuthState} />
                {this.props.children(this.state.auth)}
            </div>
        );
    }
}
