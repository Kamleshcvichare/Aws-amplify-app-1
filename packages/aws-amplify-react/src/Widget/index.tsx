export { default as Picker, IPickerProps } from './Picker';
export { default as PhotoPicker, IPhotoPickerProps } from './PhotoPicker';
export { default as TextPicker, ITextPickerProps } from './TextPicker';
export { default as SelectMFAType, ISelectMFATypeProps } from './SelectMFAType';
export { default as TOTPSetupComp, ITOTPSetupCompProps } from './TOTPSetupComp';
