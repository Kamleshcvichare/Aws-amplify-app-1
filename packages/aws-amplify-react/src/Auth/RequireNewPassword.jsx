/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';

import { I18n, JS, ConsoleLogger as Logger } from '@aws-amplify/core';
import Auth from '@aws-amplify/auth';

import AuthPiece from './AuthPiece';
import AmplifyTheme from '../AmplifyTheme';
import {
    FormSection,
    SectionHeader,
    SectionBody,
    SectionFooter,
    InputRow,
    ButtonRow,
    Link
} from '../AmplifyUI';

const logger = new Logger('RequireNewPassword');

export default class RequireNewPassword extends AuthPiece {
    constructor(props) {
        super(props);

        this._validAuthStates = ['requireNewPassword'];
        this.change = this.change.bind(this);
        this.checkContact = this.checkContact.bind(this);
    }

    checkContact(user) {
        if (!Auth || typeof Auth.verifiedContact !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }
        Auth.verifiedContact(user)
            .then(data => {
                if (!JS.isEmpty(data.verified)) {
                    this.changeState('signedIn', user);
                } else {
                    user = Object.assign(user, data);
                    this.changeState('verifyContact', user);
                }
            });
    }

    change() {
        const user = this.props.authData;
        const { password } = this.inputs;
        const { requiredAttributes } = user.challengeParam;

        let finalRequiredAttributes = {};
        let that = this;
        Object.keys(requiredAttributes).map(function (key) {

            let attributeName = requiredAttributes[key];
            finalRequiredAttributes[attributeName] = that.inputs[attributeName];

            return finalRequiredAttributes;
        })

        if (!Auth || typeof Auth.completeNewPassword !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }
        Auth.completeNewPassword(user, password, finalRequiredAttributes)
            .then(user => {
                logger.debug('complete new password', user);
                if (user.challengeName === 'SMS_MFA') {
                    this.changeState('confirmSignIn', user);
                } else if (user.challengeName === 'MFA_SETUP') {
                    logger.debug('TOTP setup', user.challengeParam);
                    this.changeState('TOTPSetup', user);
                } else {
                    this.checkContact(user);
                }

            })
            .catch(err => this.error(err));
    }

    showComponent(theme) {
        const { hide } = this.props;
        if (hide && hide.includes(RequireNewPassword)) { return null; }

        const user = this.props.authData;
        const { requiredAttributes } = user.challengeParam;
        let that = this;

        return (
            <FormSection theme={theme}>
                <SectionHeader theme={theme}>{I18n.get('Change Password')}</SectionHeader>
                <SectionBody>
                    <InputRow
                        autoFocus
                        placeholder={I18n.get('New Password')}
                        theme={theme}
                        key="password"
                        name="password"
                        type="password"
                        onChange={this.handleInputChange}
                    />
                    {
                        Object.keys(requiredAttributes).map(function (key) {

                            let value = requiredAttributes[key];
                            let displayValue = that.capitalize(value);
                            return (
                                <InputRow
                                    placeholder={I18n.get(displayValue)}
                                    theme={theme}
                                    key={value}
                                    name={value}
                                    type="text"
                                    onChange={that.handleInputChange}
                                />
                            );

                        })
                    }
                    <ButtonRow theme={theme} onClick={this.change}>
                        {I18n.get('Change')}
                    </ButtonRow>
                </SectionBody>
                <SectionFooter theme={theme}>
                    <Link theme={theme} onClick={() => this.changeState('signIn')}>
                        {I18n.get('Back to Sign In')}
                    </Link>
                </SectionFooter>
            </FormSection>
        )
    }

    capitalize(str) {
        if (str === undefined) {
            return str;
        }
        return str.replace(/\w\S*/g, function(txt){
            return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
        });
    }
}
