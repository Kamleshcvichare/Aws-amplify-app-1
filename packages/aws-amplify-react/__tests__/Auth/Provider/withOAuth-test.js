import Auth from '@aws-amplify/auth';
import * as React from 'react';
import { Component } from 'react';
import withOAuth, { OAuthButton } from '../../../src/Auth/Provider/withOAuth';
import { SignInButton, Button } from '../../../src/AmplifyUI';

describe('withOAuth test', () => {
    describe('render test', () => {
        test('render correctly', () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            const Comp = withOAuth(MockComp);
            const wrapper = shallow(<Comp/>);
            expect(wrapper).toMatchSnapshot();
        });

        test('render correctly with button', () => {
            const wrapper = shallow(<OAuthButton/>);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('signIn test', () => {
        test('happy case with connected response', () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            const spyon = jest.spyOn(Auth, 'federatedSignIn');
            const spyon2 = jest.spyOn(Auth, 'handleAuthResponse');
            
            const Comp = withOAuth(MockComp);
            const wrapper = mount(<Comp/>);
            const comp = wrapper.instance();

            comp.signIn();

            expect(spyon).toBeCalledWith(undefined);
            expect(spyon2).toBeCalled();
            spyon.mockClear();
            spyon2.mockClear();
        });

        test('Passing in a social provider', () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            const spyon = jest.spyOn(Auth, 'federatedSignIn');
            const spyon2 = jest.spyOn(Auth, 'handleAuthResponse');
            
            const Comp = withOAuth(MockComp);
            const wrapper = mount(<Comp/>);
            const comp = wrapper.instance();

            comp.signIn('Facebook');

            expect(spyon).toBeCalledWith({"provider": "Facebook"});
            expect(spyon2).toBeCalled();
            spyon.mockClear();
            spyon2.mockClear();
        });
    });
});
