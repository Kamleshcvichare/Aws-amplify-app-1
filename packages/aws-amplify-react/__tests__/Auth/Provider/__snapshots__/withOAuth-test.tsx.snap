// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOAuth test render test render correctly 1`] = `
<class_1
  OAuthSignIn={[Function]}
/>
`;

exports[`withOAuth test render test render correctly 2`] = `<div />`;

exports[`withOAuth test render test render correctly with button 1`] = `
<Button
  OAuthSignIn={[Function]}
/>
`;

exports[`withOAuth test render test render correctly with button 2`] = `
<button
  class="Button__signInButton___3bUH-"
  id="Button__oAuthSignInButton___3UGOl"
  variant="oAuthSignInButton"
>
  <span
    class="Button__signInButtonContent___xqTXJ"
  >
    Sign in with AWS
  </span>
</button>
`;
